daemons/gptp/windows/daemon_cl/windows_hal.cpp

```diff
@@ -87,12 +88,60 @@ bool WindowsTimestamper::HWTimestamper_init( InterfaceLabel *iface_label, OSNetw
 
 	if( pAdapterInfo == NULL ) return false;
 
-	if( strstr( pAdapterInfo->Description, NANOSECOND_CLOCK_PART_DESCRIPTION ) != NULL ) {
-		netclock_hz.QuadPart = NETCLOCK_HZ_NANO;
-	} else {
-		netclock_hz.QuadPart = NETCLOCK_HZ_OTHER;
+	get_phy_delay(&delay_val);
+
+	DeviceClockRateMapping *rate_map = DeviceClockRateMap;
+	while (rate_map->device_desc != NULL)
```

Comment:
Yes. The expectation is that the array has at least one terminating entry where device_desc == NULL. This is for both the PHY delay and clock rate mapping.